src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java

```diff
@@ -757,6 +809,89 @@ public void reset() {
+        getAnalyticsMessages().emptyTrackingQueues(new AnalyticsMessages.MixpanelDescription(mToken));
+        if (getPeople().isIdentified()) {
+            getPeople().deleteUser();
+            getPeople().clearCharges();
+        }
+        mPersistentIdentity.clearPreferences();
+        synchronized (mEventTimings) {
+            mEventTimings.clear();
+            mPersistentIdentity.clearTimeEvents();
+        }
+        mPersistentIdentity.clearReferrerProperties();
+        mPersistentIdentity.setOptOutTracking(true);
+    }
+
+    /**
+     * Use this method to opt-in an already opted-out user from tracking. People updates and track
+     * calls will be sent to Mixpanel after using this method.
+     * This method will internally track an opt-in event to your project. If you want to identify
+     * the opt-in event and/or pass properties to the event, see {@link #optInTracking(String)} and
+     * {@link #optInTracking(String, JSONObject)}
+     *
+     * See also {@link #optOutTracking()}.
+     */
+    public void optInTracking() {
+        optInTracking(null, null);
+    }
+
+    /**
+     * Use this method to opt-in an already opted-out user from tracking. People updates and track
+     * calls will be sent to Mixpanel after using this method.
+     * This method will internally track an opt-in event to your project.
+     *
+     * @param distinctId Optional string to use as the distinct ID for events.
+     *                   This will call {@link #identify(String)}.
+     *                   If you use people profiles make sure you manually call
+     *                   {@link People#identify(String)} after this method.
+     *
+     * See also {@link #optInTracking(String)}, {@link #optInTracking(String, JSONObject)} and
+     *  {@link #optOutTracking()}.
+     */
+    public void optInTracking(String distinctId) {
+        optInTracking(distinctId, null);
+    }
+
+    /**
+     * Use this method to opt-in an already opted-out user from tracking. People updates and track
+     * calls will be sent to Mixpanel after using this method.
+     * This method will internally track an opt-in event to your project.
+     *
+     * @param distinctId Optional string to use as the distinct ID for events.
+     *                   This will call {@link #identify(String)}.
+     *                   If you use people profiles make sure you manually call
+     *                   {@link People#identify(String)} after this method.
+     * @param properties Optional JSONObject that could be passed to add properties to the
+     *                   opt-in event that is sent to Mixpanel.
+     *
+     * See also {@link #optInTracking()} and {@link #optOutTracking()}.
+     */
+    public void optInTracking(String distinctId, JSONObject properties) {
+        mPersistentIdentity.setOptOutTracking(false);
+        if (distinctId != null) {
+            identify(distinctId);
+        }
+        track("$opt_in", properties);
```

Comment:
you guys probably have discussed this before but please make sure we send the same event in all the SDKs.